This patch moves to the front of the loop the bounds verification, otherwise
some files trigger invalid memory accesses.

--- a/src/ole.c
+++ b/src/ole.c
@@ -331,6 +331,23 @@
 			return NULL;
 		}
 		while(1) {
+			if(chainCurrent < 0 ||
+			   chainCurrent >= (
+                               e->isBigBlock ?
+                               ((bbdNumBlocks*sectorSize)/4) :
+                               ((sbdNumber*shortSectorSize)/4)
+                               ) ||
+			   (e->numOfBlocks >
+				e->length/(
+                                    e->isBigBlock ?
+                                    sectorSize :
+                                    shortSectorSize
+                                    )
+                                )
+                           ) {
+/*   				fprintf(stderr, "chain End=%ld\n", chainCurrent);   */
+				break;
+			}
 /* 			fprintf(stderr, "chainCurrent=%ld\n", chainCurrent); */
 			e->blocks[e->numOfBlocks++] = chainCurrent;
 			if (e->numOfBlocks >= chainMaxLen) {
@@ -353,15 +370,6 @@
 			} else {
 				chainCurrent=-1;
 			}
-			if(chainCurrent <= 0 ||
-			   chainCurrent >= ( e->isBigBlock ?
-								 ((bbdNumBlocks*sectorSize)/4)
-								 : ((sbdNumber*shortSectorSize)/4) ) ||
-			   (e->numOfBlocks >
-				e->length/(e->isBigBlock ? sectorSize : shortSectorSize))) {
-/*   				fprintf(stderr, "chain End=%ld\n", chainCurrent);   */
-				break;
-			}
 		}
 	}
 	
